### PR TITLE
shield generators no longer gib mobs

### PR DESCRIPTION
**Pull request by zeskorion**
## About The Pull Request
Shield generators no longer instantly gib mobs as soon as they are turned on, instead applying an electric shock.

## Why It's Good For The Game
Removes a method of cheep instagibbing on targets. This behavior is unintuitive (nothing else in the game acts this way, the closest being that doors crush you when closed), and instant gibs are cheap. The shield generator doesnt break other objects in its path, why should it be able to kill mobs instantly?

## Testing Photographs and Procedure
not necessary

## Changelog
:cl:
balance: shield generators no longer gib mobs when activated. instead, they apply an electric shock
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
Time to make a counter PR that makes shield generators destroy everything in their path

**Comment by zeskorion** [comment]
> Time to make a counter PR that makes shield generators destroy everything in their path

now its time to nerf the cremator as well uwu

File: code/game/machinery/shieldgen.dm [hunk 1]
```diff
@@ -419,8 +419,8 @@
 		needs_power = TRUE
 		setDir(get_dir(gen_primary, gen_secondary))
 	for(var/mob/living/L in get_turf(src))
-		visible_message("<span class='danger'>\The [src] is suddenly occupying the same space as \the [L]!</span>")
-		L.gib()
+		visible_message("<span class='danger'>\The [src] zaps through \The [L]!</span>")
+		L.electrocute_act(30, src, 1, FALSE, FALSE, FALSE, FALSE, TRUE)
```
Comment (MNarath1) [inline code/game/machinery/shieldgen.dm]:
```suggestion
		L.electrocute_act(30, src)
```
Litterally all the other arguments you put in there are their default values anyways except the first two obciously.

**Review by MNarath1 — CHANGES_REQUESTED** [review]

File: code/game/machinery/shieldgen.dm (hunk 1 above)
Comment (MNarath1) [reply code/game/machinery/shieldgen.dm]: ~~also i just realized there was a FALSE to many in there so the TRUE would have not aligned up with the stun argument~~

Comment (MNarath1) [reply code/game/machinery/shieldgen.dm]: nope i just counted wrong  still they are not needed when thats their default anyways

File: code/game/machinery/shieldgen.dm [hunk 2]
```diff
@@ -419,8 +419,8 @@
 		needs_power = TRUE
 		setDir(get_dir(gen_primary, gen_secondary))
 	for(var/mob/living/L in get_turf(src))
-		visible_message("<span class='danger'>\The [src] is suddenly occupying the same space as \the [L]!</span>")
-		L.gib()
+		visible_message("<span class='danger'>\The [src] zaps through \The [L]!</span>")
```
Comment (PowerfulBacon) [inline code/game/machinery/shieldgen.dm]: Why is the now capitalized (and does that even matter in byond, I assume it does).
Also [L] will automatically insert the anyway.

File: code/game/machinery/shieldgen.dm [hunk 3]
```diff
@@ -419,8 +419,8 @@
 		needs_power = TRUE
 		setDir(get_dir(gen_primary, gen_secondary))
 	for(var/mob/living/L in get_turf(src))
-		visible_message("<span class='danger'>\The [src] is suddenly occupying the same space as \the [L]!</span>")
-		L.gib()
+		visible_message("<span class='danger'>\The [src] zaps through [L]!</span>")
```
Comment (PowerfulBacon) [inline code/game/machinery/shieldgen.dm:422]: I still think this doesn't really make sense.
The burnard silkwind zaps through the shield wall?

Comment (ike709) [reply code/game/machinery/shieldgen.dm:422]: The text macro makes this fine @PowerfulBacon

**Comment by Mat05usz** [comment]
And that PR serves what purpose exactly besides just being a randomly picked removal?
What is gonna happen will be a shittone of mobs being spawned in xenobio, with no way of removing them thus further adding to the creeping late round lag.

Edit: 
Before you bring the argument of "insta killing players so bad, why is in the game???" I'm just gonna tell you that in my 2500 hours of playtime as living, I was gibbed by xenobio exactly once using that shield, and another time being a resourceful antag setting a trap. It's literally a non existent problem that you're trying to remove while causing a lot more damage elsewhere.

Unless it's an ided PR, then I have no questions.

**Comment by zeskorion** [comment]
> And that PR serves what purpose exactly besides just being a randomly picked removal? What is gonna happen will be a shittone of mobs being spawned in xenobio, with no way of removing them thus further adding to the creeping late round lag.

maybe dont spawn a shitton of mobs in xenobio 🤔 
that in and of itself is a problem that can be dealt with if this makes it a problem 👉😏 👉  

> Edit: Before you bring the argument of "insta killing players so bad, why is in the game???" I'm just gonna tell you that in my 2500 hours of playtime as living, I was gibbed by xenobio exactly once using that shield, and another time being a resourceful antag setting a trap. It's literally a non existent problem that you're trying to remove while causing a lot more damage elsewhere.

having a positional, reusable instagib is always bad

File: code/game/machinery/shieldgen.dm (hunk 3 above)
Comment (PowerfulBacon) [reply code/game/machinery/shieldgen.dm:422]: the verb just doesn't make sense

Comment (zeskorion) [reply code/game/machinery/shieldgen.dm:422]: You have it the wring way around
"The shield wall zaps through Johnny Cupcakes"

**Comment by Isy232** [comment]
Nah, its funny because it actually requires some thinking to use effectively unlike your generic desword or CQC man, both of which are also far harder to counter.
Malf AI + BS launchpads
If an antag is smart, maybe they deserve success?

**Comment by Archanial** [comment]
🔑

**Comment by PowerfulBacon** [comment]
🔑

**Pull request closed without merging**